# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1445578222 -7200
#      Fri Oct 23 07:30:22 2015 +0200
# Node ID 98a730e0dd5135acbf668d9bf44c4d4eca86cbc5
# Parent  fb0f233d7780c83553b14708357e33d7be9fc1ba
remove column visibility of file browser from settings dialog

* settings-dialog.cc (settings_dialog): remove initializations of obsolete
  dialog items;
  (write_changed_settings): remove writing settings from obsolete dialog items

* settings-dialog.ui: Remove items related to the display of the file browser

diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -491,26 +491,16 @@ settings_dialog::settings_dialog (QWidge
   if (cursorType == "ibeam")
     ui->terminal_cursorType->setCurrentIndex (0);
   else if (cursorType == "block")
     ui->terminal_cursorType->setCurrentIndex (1);
   else if (cursorType == "underline")
     ui->terminal_cursorType->setCurrentIndex (2);
 
   // file browser
-  ui->showFileSize->setChecked (
-    settings->value ("filesdockwidget/showFileSize", false).toBool ());
-  ui->showFileType->setChecked (
-    settings->value ("filesdockwidget/showFileType", false).toBool ());
-  ui->showLastModified->setChecked (
-    settings->value ("filesdockwidget/showLastModified",false).toBool ());
-  ui->showHiddenFiles->setChecked (
-    settings->value ("filesdockwidget/showHiddenFiles",false).toBool ());
-  ui->useAlternatingRowColors->setChecked (
-    settings->value ("filesdockwidget/useAlternatingRowColors",true).toBool ());
   connect (ui->sync_octave_directory, SIGNAL (toggled (bool)),
            this, SLOT (set_disabled_pref_file_browser_dir (bool)));
   ui->sync_octave_directory->setChecked (
     settings->value ("filesdockwidget/sync_octave_directory",true).toBool ());
   ui->cb_restore_file_browser_dir->setChecked (
     settings->value ("filesdockwidget/restore_last_dir",false).toBool ());
   ui->le_file_browser_dir->setText (
     settings->value ("filesdockwidget/startup_dir").toString ());
@@ -835,26 +825,17 @@ settings_dialog::write_changed_settings 
   settings->setValue ("editor/create_new_file",
                       ui->editor_create_new_file->isChecked ());
   settings->setValue ("editor/always_reload_changed_files",
                       ui->editor_reload_changed_files->isChecked ());
   settings->setValue ("terminal/fontSize", ui->terminal_fontSize->value ());
   settings->setValue ("terminal/fontName",
                       ui->terminal_fontName->currentFont ().family ());
 
-  settings->setValue ("filesdockwidget/showFileSize",
-                      ui->showFileSize->isChecked ());
-  settings->setValue ("filesdockwidget/showFileType",
-                      ui->showFileType->isChecked ());
-  settings->setValue ("filesdockwidget/showLastModified",
-                      ui->showLastModified->isChecked ());
-  settings->setValue ("filesdockwidget/showHiddenFiles",
-                      ui->showHiddenFiles->isChecked ());
-  settings->setValue ("filesdockwidget/useAlternatingRowColors",
-                      ui->useAlternatingRowColors->isChecked ());
+  // file browser
   settings->setValue ("filesdockwidget/sync_octave_directory",
                       ui->sync_octave_directory->isChecked ());
   settings->setValue ("filesdockwidget/restore_last_dir",
                       ui->cb_restore_file_browser_dir->isChecked ());
   settings->setValue ("filesdockwidget/startup_dir",
                       ui->le_file_browser_dir->text ());
 
 
diff --git a/libgui/src/settings-dialog.ui b/libgui/src/settings-dialog.ui
--- a/libgui/src/settings-dialog.ui
+++ b/libgui/src/settings-dialog.ui
@@ -27,17 +27,17 @@
     <widget class="QTabWidget" name="tabWidget">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="currentIndex">
-      <number>1</number>
+      <number>4</number>
      </property>
      <widget class="QWidget" name="tab_general">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <attribute name="title">
        <string>General</string>
       </attribute>
@@ -46,18 +46,18 @@
         <widget class="QScrollArea" name="scrollArea_2">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_2">
           <property name="geometry">
            <rect>
             <x>0</x>
-            <y>-4</y>
-            <width>662</width>
+            <y>0</y>
+            <width>570</width>
             <height>382</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_17">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="title">
               <string>Interface</string>
@@ -1456,18 +1456,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_4">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>488</width>
+            <height>236</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <layout class="QGridLayout" name="gridLayout_7">
                <property name="sizeHint" stdset="0">
@@ -1750,67 +1750,16 @@
             <x>0</x>
             <y>0</y>
             <width>678</width>
             <height>378</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_18">
            <item>
-            <widget class="QGroupBox" name="groupBox_3">
-             <property name="title">
-              <string>Display</string>
-             </property>
-             <layout class="QVBoxLayout" name="verticalLayout_23">
-              <item>
-               <layout class="QGridLayout" name="gridLayout_11">
-                <property name="topMargin">
-                 <number>0</number>
-                </property>
-                <item row="1" column="0">
-                 <widget class="QCheckBox" name="showFileType">
-                  <property name="text">
-                   <string>Show file type</string>
-                  </property>
-                 </widget>
-                </item>
-                <item row="3" column="0">
-                 <widget class="QCheckBox" name="showHiddenFiles">
-                  <property name="text">
-                   <string>Show hidden files</string>
-                  </property>
-                 </widget>
-                </item>
-                <item row="0" column="0">
-                 <widget class="QCheckBox" name="showFileSize">
-                  <property name="text">
-                   <string>Show file size</string>
-                  </property>
-                 </widget>
-                </item>
-                <item row="2" column="0">
-                 <widget class="QCheckBox" name="showLastModified">
-                  <property name="text">
-                   <string>Show date of last modification</string>
-                  </property>
-                 </widget>
-                </item>
-                <item row="4" column="0">
-                 <widget class="QCheckBox" name="useAlternatingRowColors">
-                  <property name="text">
-                   <string>Alternating row colors</string>
-                  </property>
-                 </widget>
-                </item>
-               </layout>
-              </item>
-             </layout>
-            </widget>
-           </item>
-           <item>
             <widget class="QGroupBox" name="groupBox_4">
              <property name="title">
               <string>Behavior</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_24">
               <item>
                <layout class="QGridLayout" name="gridLayout_8">
                 <item row="4" column="0">
@@ -1921,18 +1870,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_6">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>203</width>
+            <height>79</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_19">
            <item>
             <layout class="QGridLayout" name="gridLayout_14">
              <property name="topMargin">
               <number>0</number>
              </property>
@@ -1990,18 +1939,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_8">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>368</width>
+            <height>244</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_25">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_26">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
@@ -2253,18 +2202,18 @@
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents_7">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
-            <width>678</width>
-            <height>378</height>
+            <width>529</width>
+            <height>204</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_20">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <widget class="QCheckBox" name="checkbox_allow_web_connect">
                <property name="text">
